packages/app/provider/WalletConnect.tsx

```diff
@@ -0,0 +1,256 @@
+  signAndSubmitXdr: (_xdr: string) => Promise<null | unknown>;
+}
+
+export const ClientContext = createContext<IContext>({} as IContext);
+
+export function ClientContextProvider({
+  children,
+}: {
+  children: ReactNode | ReactNode[];
+}) {
+  const [client, setClient] = useState<Client>();
+  const [session, setSession] = useState<SessionTypes.Struct>();
+  const [accounts, setAccounts] = useState<string[]>([]);
+
+  const reset = () => {
+    setSession(undefined);
+    setAccounts([]);
+  };
+
+  const onSessionConnected = useCallback((_session: SessionTypes.Struct) => {
+    const allNamespaceAccounts = Object.values(_session.namespaces)
+      .map((namespace) => namespace.accounts)
+      .flat();
+
+    setSession(_session);
+    setAccounts(allNamespaceAccounts);
+  }, []);
+
+  const connect = useCallback(async () => {
+    if (typeof client === "undefined") {
+      throw new Error("WalletConnect is not initialized");
+    }
+
+    try {
+      const requiredNamespaces = {
+        stellar: {
+          methods: ["stellar_signAndSubmitXDR", "stellar_signXDR"],
+          chains: ["stellar:pubnet", "stellar:testnet"],
+          events: [],
+        },
+      };
+
+      const { uri, approval } = await client.connect({
+        pairingTopic: undefined,
+        requiredNamespaces,
+      });
+
+      if (!uri) return;
+      if (Platform.OS === "web") {
+        QRCodeModal.open(
+          uri,
+          () => {
+            console.log("EVENT", "QR Code Modal closed");
+          },
+          {
+            desktopLinks: [],
+            mobileLinks: ["lobstr"],
+          }
+        );
+      } else {
+        Alert.alert("WalletConnect link", uri, [
+          {
+            text: "Cancel",
+            onPress: () => console.log("Cancel Pressed"),
+            style: "cancel",
+          },
+          {
+            text: "Copy",
+            onPress: async () => {
+              await Clipboard.setStringAsync(uri);
+            },
+          },
+        ]);
+        // TODO find out proper deep link
+        // Linking.openURL("https://lobstr.co/uni");
+      }
+
+      const session = await approval();
+      console.log("Established session:", session);
+      await onSessionConnected(session);
+      if (Platform.OS === "web") {
+        QRCodeModal.close();
+      }
+      return session;
+    } catch (e) {
+      if (Platform.OS === "web") {
+        QRCodeModal.close();
+      }
+      console.error(e);
+    }
+  }, [client, onSessionConnected]);
+
+  const disconnect = useCallback(async () => {
+    if (typeof client === "undefined") {
+      throw new Error("WalletConnect is not initialized");
+    }
+    if (typeof session === "undefined") {
+      throw new Error("Session is not connected");
+    }
+    await client.disconnect({
+      topic: session.topic,
+      reason: getSdkError("USER_DISCONNECTED"),
+    });
+    reset();
+  }, [client, session]);
+
+  const _subscribeToEvents = useCallback(
```

Comment:
Forgot to ask, why are we using underscores at the beginning of names?

Reply:
it was meant to indicate that some functions should be private, but I guess it might be confusing. I'll change that.